### PR TITLE
env: override default scheme for Apple Python venv

### PR DESCRIPTION
**Pull request by layday**

### CODE REVIEW

File: src/build/env.py [hunk 1]
```diff
@@ -246,20 +246,17 @@ def _find_executable_and_scripts(path):  # type: (str) -> Tuple[str, str, str]
...
-        exe = 'pypy3' if platform.python_implementation() == 'PyPy' else 'python'
-        if os.name == 'nt':
-            exe = '{}.exe'.format(exe)
-        executable = os.path.join(env_scripts, exe)
+        is_apple_python = 'osx_framework_library' in sysconfig.get_scheme_names()
+        paths = (
+            sysconfig.get_paths(scheme='posix_prefix', vars=config_vars)
+            if is_apple_python
+            else sysconfig.get_paths(vars=config_vars)
+        )
```
Comment (FFY00) [inline src/build/env.py:254]: Personally, I think the following is clearer:

```suggestion
        if 'osx_framework_library' in sysconfig.get_scheme_names():  # apple python
            paths = sysconfig.get_paths(scheme='posix_prefix', vars=config_vars)
        else:
            paths = sysconfig.get_paths(vars=config_vars)
```

Comment (FFY00) [reply src/build/env.py:254]: Maybe add a small comment there too explaining this.

Comment (layday) [reply src/build/env.py:254]: Sure, that works.

**Comment by layday** [comment]
Squashed.

**Review by FFY00 — APPROVED** [review]

**Comment by FFY00** [comment]
@gaborbernat could you have a look before merging?

**Review by henryiii — APPROVED** [review]

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**